client/src/pages/Challenge.tsx

```diff
@@ -1,118 +1,181 @@
+                  {details.keyPatterns.map((pattern, index) => {
+                    return (
+                      <li key={index}>
+                        {pattern + (pattern[-1] !== "." && ".")}
+                      </li>
+                    );
+                  })}
                 </ul>
-            </Col>
-            :
-            <Col>
+              </Col>
+            ) : (
+              <Col>
                 <ButtonToolbar className="d-flex align-items-end justify-content-around">
-                    <Button className="m-1" target="_blank" href="/editor">Open Editor</Button>
-                    <Button className="m-1" href={"/solutions/post/" + id}>Post a Solution</Button>
-                    <Button className="m-1" target="_blank" href={"/solutions/challenge/" + id}>View Solutions</Button>
+                  <Button className="m-1" target="_blank" href="/editor">
+                    Open Editor
+                  </Button>
+                  <Button className="m-1" href={"/solutions/post/" + id}>
+                    Post a Solution
+                  </Button>
+                  <Button
+                    className="m-1"
+                    target="_blank"
+                    href={"/solutions/challenge/" + id}
+                  >
+                    View Solutions
+                  </Button>
                 </ButtonToolbar>
-            </Col>
-            }            
+              </Col>
+            )
+          }
         </Row>
         {
-        // If completed, display buttons at the bottom
-        completed &&
-        <Row>
-           <ButtonToolbar className="d-flex align-items-end justify-content-around">
-                <Button className="m-1" target="_blank" href="/editor">Open Editor</Button>
-                <Button className="m-1" href={"/solutions/post/" + id}>Post a Solution</Button>
-                <Button className="m-1" target="_blank" href={"/solutions/challenge/" + id}>View Solutions</Button>
-            </ButtonToolbar>
-        </Row>}
-        </ section>
-    </Container>);
-  };
-  
-  export default Challenge;
-  
\ No newline at end of file
+          // If completed, display buttons at the bottom
+          completed && (
+            <Row>
+              <ButtonToolbar className="d-flex align-items-end justify-content-around">
+                <Button className="m-1" target="_blank" href="/editor">
+                  Open Editor
+                </Button>
+                <Button className="m-1" href={"/solutions/post/" + id}>
+                  Post a Solution
+                </Button>
+                <Button
+                  className="m-1"
+                  target="_blank"
+                  href={"/solutions/challenge/" + id}
+                >
+                  View Solutions
+                </Button>
+              </ButtonToolbar>
+            </Row>
+          )
+        }
+      </section>
+    </Container>
+  );
+};
+
+export default Challenge;
```

Comment:
Looks good. I like that you used headings h1 h2 h3 in hierarchal order as this helps improve navigation for screen readers.